pkg/controllers/v1alpha1/dataset/dataset_controller.go

```diff
@@ -180,7 +180,15 @@ func (r *DatasetReconciler) reconcileDatasetDeletion(ctx reconcileRequestContext
 		return utils.RequeueAfterInterval(time.Duration(10 * time.Second))
 	}
 
-	// 2. Remove finalizer
+	// 2. if there are datasets mounted this dataset, then requeue
+	if ctx.Dataset.Status.DatasetRef != nil && len(ctx.Dataset.Status.DatasetRef) > 0 {
+		ctx.Log.Error(err, "Failed to delete dataset with reference", "DatasetDeleteError", ctx)
+		r.Recorder.Eventf(&ctx.Dataset, v1.EventTypeWarning, common.ErrorDeleteDataset,
+			"Failed to delete dataset because there are datasets mounted to it")
```

Comment:
Could you add information about how to checking the related mounted datasets so the user can debug it?